### PR TITLE
CNFT1-2198 Segmented Buttons

### PR DESCRIPTION
**Pull request by jlee-enquizit**
## Description

Here we create a new SegmentedButtons component to align the styling with our design system.
image

## Tickets

* [CNFT1-2198](https://cdc-nbs.atlassian.net/browse/CNFT1-2198)
* [CNFT1-2199](https://cdc-nbs.atlassian.net/browse/CNFT1-2199)
* [CNFT1-2200](https://cdc-nbs.atlassian.net/browse/CNFT1-2200)
* [CNFT1-2201](https://cdc-nbs.atlassian.net/browse/CNFT1-2201)

## Checklist before requesting a review
- [ ] PR focuses on a single story
- [x] Code has been fully tested to meet acceptance criteria
- [x] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [x] All new functions/classes/components reasonably small
- [x] Functions/classes/components focused on one responsibility
- [x] Code easy to understand and modify (clarity over concise/clever)
- [x] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [x] PR does not contain hardcoded values (Uses constants)
- [x] All code is covered by unit or feature tests

[CNFT1-2198]: https://cdc-nbs.atlassian.net/browse/CNFT1-2198?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CNFT1-2199]: https://cdc-nbs.atlassian.net/browse/CNFT1-2199?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CNFT1-2200]: https://cdc-nbs.atlassian.net/browse/CNFT1-2200?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CNFT1-2201]: https://cdc-nbs.atlassian.net/browse/CNFT1-2201?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/modernization-ui/src/components/SegmentedButtons/SegmentedButtons.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    onBlur?: () => void;
+    onChange?: () => void;
+    onClick?: (button: any) => void;
+    value: string | number;
+};
+
+export const SegmentedButtons = ({ buttons, value, onBlur, onChange, onClick }: Props) => {
+    return (
+        <div className={styles.segmentedButtons}>
+            {buttons.map((button: any, key: number) => (
```
Comment (mpeels) [inline apps/modernization-ui/src/components/SegmentedButtons/SegmentedButtons.tsx:14]: Issue: use of `any`

File: apps/modernization-ui/src/components/SegmentedButtons/SegmentedButtons.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { Button, Icon } from '@trussworks/react-uswds';
+import styles from './segmentedButtons.module.scss';
+type Props = {
+    buttons: any;
```
Comment (mpeels) [inline apps/modernization-ui/src/components/SegmentedButtons/SegmentedButtons.tsx:4]: Issue: use of any

**Review by mpeels — CHANGES_REQUESTED** [review]

**Comment by mpeels** [comment]
https://cdc-nbs.atlassian.net/browse/CNFT1-2200 states the SegmentedButtons should be limited to 2-4 entries.

**Review by mpeels — DISMISSED** [review]

File: apps/modernization-ui/src/components/SegmentedButtons/segmentedButtons.module.scss [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        }
+        &:hover {
+            box-shadow: none;
+            border-bottom: 0.19rem solid colors.$primary;
+        }
+        &:not([disabled]):focus{
+            outline: none;
+        }
+        &:first-of-type {
+            border-radius: 0.3rem 0 0 0.3rem;
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/components/SegmentedButtons/segmentedButtons.module.scss:28]: nit:  The design system specifies a `border-radius` of `0.3125rem`
_this may be the most nit picky review I've ever written_

**Review by adamloup-enquizit — APPROVED** [review]
👍 <sup> 👍 <sup> 👍 </sup></sup>

**Review by hclarkEnq — APPROVED** [review]

**Review by mpeels — APPROVED** [review]

**Pull request merged**